### ISSUE TITLE
Low battery alert will not stop

### ISSUE THREAD

**Issue by hugo-id** [pop-os/cosmic-epoch#1279]
I'm running a Framework laptop with Cosmic Epoch:
![screenshot-2024-11-24-07-33-25](https://github.com/user-attachments/assets/7d2a204f-6904-45e1-9132-6f37ec246f61)

When the low battery sound starts, it doesn't stop!

Every 4 seconds, it sounds again, and again, and again.

That's _incredibly_ annoying, but maybe I can get it when battery is critical.

The **bug** is that, even after charging, the alert doesn't stop until I reboot my laptop.

**Comment by Department-stack** [comment]
That is annoying, I personally dont even hear a crictical battery warning sound on low battery

**Comment by timaeos** [comment]
This is related to `cosmic-settings-daemon` it seems. I am having this issue and had to kill `cosmic-settings-daemon` to get it to stop. It might be worth making an issue there instead. 
I haven't looked at the code yet since I don't know rust but hopefully it's something simple. I'd expect it to be an issue how battery level is being reported.

**Comment by x86lover** [comment]
I can also confirm this... Running Pop_OS! 24.04 (Alpha 4) on my Thinkpad L390. Thankfully mine stops when charging.

**Comment by russdam** [comment]
Yip also getting it on Arch with Alpha4, terrible clack-clack click-click continuously when I reach 100%.

**Comment by aanderson6** [comment]
I can report that on my thinkpad t480 I have a similar issue where the battery alerts go off until the battery is above the threshold of 10% after which it stops. But this happens even if I plug in the device, whereas the alerts should turn off after I plug it in even if it isn't at 10%

**Comment by lucille-bellepleure** [comment]
Also reporting my Xiaomi 14 airbook has the same issue

**Comment by atiq-cs** [comment]
You guys are getting low battery alert. Mine never gets that as a result often battery drains to 0 as I don't notice. I wonder how to enable low battery alert on mine. I am on ProArt P16 (AMD) model.

Update: this seems fixed after moving to Pop OS from fedora. However, logically it should still be fixed for fedora. If someone can test that will be great!

**Comment by wash2** [comment]
After installing https://github.com/pop-os/cosmic-settings-daemon/commit/8616c40d235164779cd3f2ceec1fe9b2b4aceb40 do you still experience this issue?

**Comment by TheDarkThief** [comment]
It seemed to still experience this issue even with 8616c40 but I'll update and give you an update tomorrow

**Comment by Zarelark** [comment]
Also experiencing this. Incredibly annoying.

**Comment by wash2** [comment]
Ok, thanks. I'll look into this issue some more soon.

**Comment by mohdomi** [comment]
I am also experiencing the same and it is really really annoying. please fix it as soon as possible. mine is a lenovo ideapad gaming 3 (2022) model.

**Comment by nano-o** [comment]
I'm having the same problem on a brand new lemur pro. Killing cosmic-settings-daemon stops it.

**Comment by danielpza** [comment]
Recently installed PopOS 24.04 and I'm having this same issue. As others said, super annoying. On a lemur pro. Killing the cosmic-settings-daemon does stop it `pkill -f cosmic-settings-daemon`

**Comment by danielpza** [comment]
Is there a way to prevent the warning entirely without having to kill the cosmic-settings-daemon?

**Comment by bachriotmane** [comment]
same issue here ! any updates ?

**Comment by JuergenFleiss** [comment]
Please fix this.

**Comment by nicwhites** [comment]
dudes... why is this not a simple option in notifications section in settings...

**Comment by pineapplemachine** [comment]
Egad, there is bad UX and then there is whatever this is. No I do not want my laptop repeatedly playing an incredibly irritating ticking sound ever, but especially not while I still have 30 minutes of battery life yet. How is there no way to turn this off? How is this even a thing in the first place?

**Comment by leviport** [comment]
This issue was originally tracking a bug where the low battery alert would trigger and not stop until the battery was charged back up above the threshold. That was addressed, making the alert stop when the charger was connected. 

So I believe that now, this issue is being used to track that the threshold is too high for some users/laptops?

**Comment by pineapplemachine** [comment]
I would describe the issue as the fact that this is happening at all. I would think this can could be an opt-in if some users want this, but otherwise displaying an alert and playing a sound ONE TIME is quite enough for me, and I think everyone else still commenting here.

**Comment by leviport** [comment]
Running a battery to 0% not only poses a risk of losing unsaved work, it also causes a lot of wear and tear on batteries. However we are looking into making the alert threshold configurable and the alert dismissable.

**Comment by JuergenFleiss** [comment]
> Running a battery to 0% not only poses a risk of losing unsaved work, it also causes a lot of wear and tear on batteries. However we are looking into making the alert threshold configurable and the alert dismissable.

Yes please, sometimes you are presenting or have to work and just use those last few percent and want to do so without a constant alarm :)

**Comment by chris-010** [comment]
**Root cause** (in `cosmic-settings-daemon`, not `cosmic-osd`).

The repeating alarm is `critical_battery_nag()` in `src/battery.rs`. It plays the `battery-low` sound every 3s until it receives a stop signal (`false`) on its channel. That `false` is only ever sent when **(1)** an acpid AC-plug *Plugged* event arrives while critical, or **(2)** the battery percentage climbs back to ≥10%.

While charging below 10%, the `percent < 10.0` handler takes `if current_battery == Critical { continue }` and never re-checks the charging state, so cancellation depends entirely on the acpid plug event. On hardware where acpid connects (so the alarm *does* fire) but never emits the `ac_adapter` event — Framework and others in this thread — path (1) never happens, leaving only path (2): the alarm stops at 10%.

Confirmed on Pop!_OS 24.04 / COSMIC: at 7% with the charger connected, UPower reports `State = Charging` and `WarningLevel = None`, yet the daemon keeps nagging because it tracks its own `current_battery = Critical` and ignores UPower's state.

**Fix:** drive the nag off UPower's `State` (already available via the daemon's display-device proxy) — cancel it as soon as the device stops discharging (a new `receive_state_changed` arm) and re-evaluate on each percentage tick instead of `continue`-ing.

PR: pop-os/cosmic-settings-daemon#161
